pkg/controller/snapshotter/snapshotter.go

```diff
@@ -705,33 +828,18 @@ func (vs *volumeSnapshotter) UpdateVolumeSnapshot(snapshotName string, status *[
 		return nil, fmt.Errorf("Error: expecting type VolumeSnapshot but received type %T", objCopy)
 	}
 
-	// VolumeSnapshot Metadata Labels not found. Update VolumeSnapshot by creating the Metadata Lables...
-	if bFoundVolumeSnapshotMetadata == false {
-		tags := make(map[string]string)
-		tags["Timestamp"] = fmt.Sprintf("%d", time.Now().UnixNano())
-		snapshotCopy.Metadata.Labels = tags
-		glog.Infof("UpdateVolumeSnapshot [%s]: Metadata Name: %s Metadata Namespace: %s Setting tags in Metadata Labels: [%#v].", snapshotName, snapshotCopy.Metadata.Name, snapshotCopy.Metadata.Namespace, snapshotCopy.Metadata.Labels)
+	snapshotDataObj := vs.getSnapshotDataFromSnapshotName(snapshotName)
+	if snapshotDataObj == nil {
+		glog.Infof("UpdateVolumeSnapshot: VolumeSnapshotData not created for snapshot %s yet.", snapName)
 	} else {
-		glog.Infof("UpdateVolumeSnapshot: Setting VolumeSnapshotData name in VolumeSnapshotSpec of VolumeSnapshot object")
-		snapshotDataObj := vs.getSnapshotDataFromSnapshotName(snapshotName)
-		if snapshotDataObj == nil {
-			//TODO(xyang): Should we return error here?
-			return nil, fmt.Errorf("Error getting VolumeSnapshotData for VolumeSnapshot %s", snapshotName)
-		}
-
+		glog.Infof("UpdateVolumeSnapshot: Setting VolumeSnapshotData name %s in VolumeSnapshot object %s", snapshotDataObj.Metadata.Name, snapName)
 		snapshotCopy.Spec.SnapshotDataName = snapshotDataObj.Metadata.Name
-		/*snapshotCopy.Status.Conditions = []crdv1.VolumeSnapshotCondition{
-			{
-				Type:               crdv1.VolumeSnapshotConditionPending,//crdv1.VolumeSnapshotConditionReady,
-				Status:             v1.ConditionTrue,
-				Message:            "Snapshot created succsessfully",
-				LastTransitionTime: metav1.Now(),
-			},
-		}*/
-		if status != nil {
-			glog.Infof("UpdateVolumeSnapshot: Setting Status in VolumeSnapshot object")
-			snapshotCopy.Status.Conditions = *status
-		}
+	}
+
+	if status != nil {
+		glog.Infof("UpdateVolumeSnapshot: Setting status in VolumeSnapshot object.")
+		ind := len(*status) - 1
+		snapshotCopy.Status.Conditions = append(snapshotCopy.Status.Conditions, (*status)[ind])
```

Comment:
Do we really want to append the conditions array? For example when testing in AWS I can see two "Pending" conditions and then a "Ready" one. I would simply replace the conditions with the last known status.

Reply:
I'm not sure.  We should at least check the last existing condition and only append the new one if it is different.

I see in replicaset, it is appending the condition: 
https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/replicaset/replica_set_utils.go#L158

In PersistentVolume controller, however, PersistentVolumePhase is used which is just one string such as "Pending" or "Available".